src/sentry/testutils/helpers/features.py

```diff
@@ -141,7 +142,73 @@ def batch_features_override(
             yield
 
 
-with_feature = Feature
+def with_feature(names: str | Iterable[str] | dict[str, bool]):
+    """
+    Control whether a feature is enabled.
+
+    Can be used as:
+    - Context manager: with with_feature('feature-name'):
+    - Method decorator: @with_feature('feature-name')
+    - Class decorator: @with_feature('feature-name')
+    """
+
+    class FeatureContextManagerOrDecorator:
```

Comment:
Does this class definition need to be nested within this function? I ask because we have a similar pattern with our Silo test decorators that doesn't nest this way: https://github.com/getsentry/sentry/blob/049801c70a2d753319e7631da7ac9c9a8de35720/src/sentry/testutils/silo.py#L185

Seems like it might be a bit cleaner to separate the function from the class definition if we can.